File: face.c
#include <opencv2/opencv.hpp>
#include <opencv2/face.hpp>
#include <vector>

using namespace cv;
using namespace cv::face;

int main() {
    // Load pre-trained face detector
    CascadeClassifier faceDetector("haarcascade_frontalface_default.xml");

    // Load the image
    Mat image = imread("your_image.jpg");

    // Convert to grayscale
    Mat gray;
    cvtColor(image, gray, COLOR_BGR2GRAY);

    // Detect faces
    std::vector<Rect> faces;
    faceDetector.detectMultiScale(gray, faces);

    // Initialize face recognizer
    Ptr<LBPHFaceRecognizer> recognizer = LBPHFaceRecognizer::create();
    
    // Train the recognizer (Load the dataset and fit the model)

    // For each face detected, recognize it
    for (Rect face : faces) {
        Mat faceROI = gray(face);
        int label;
        double confidence;

        recognizer->predict(faceROI, label, confidence);
        // Draw rectangle around the face and label it with confidence
    }

    // Display the result
    imshow("Face Recognition", image);
    waitKey(0);

    return 0;
}
 

File: face.py
import cv2 
import face_recognition 

# Load an image
image = face_recognition.load_image_file("your_image.jpg")

# Find all face locations
face_locations = face_recognition.face_locations(image)
print(f"Found {len(face_locations)} face(s) in this photograph.")

# Display the image with face locations
for face_location in face_locations:
    top, right, bottom, left = face_location
    cv2.rectangle(image, (left, top), (right, bottom), (0, 255, 0), 2)

# Convert the image to BGR color space
image = cv2.cvtColor(image, cv2.COLOR_RGB2BGR)

# Show the image with faces recognized
cv2.imshow("Faces Found", image)
cv2.waitKey(0)
cv2.destroyAllWindows()
